components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -168,11 +169,39 @@ BraveProfileSyncServiceImpl::BraveProfileSyncServiceImpl(Profile* profile,
     : BraveProfileSyncService(std::move(init_params)),
       brave_sync_client_(BraveSyncClient::Create(this, profile)) {
   brave_sync_words_ = std::string();
-  brave_sync_prefs_ =
+  brave_sync_prefs_disk_store_ =
       std::make_unique<prefs::Prefs>(sync_client_->GetPrefService());
+  brave_sync_prefs_ = brave_sync_prefs_disk_store_.get();
+
+  if (IsSyncSetupIncomplete()) {
```

Comment:
having `IsBraveSyncInitialized`, `IsBraveSyncConfigured` and `IsSyncSetupIncomplete` is just way too confusing. We need to simplify the possible states and add comments to explain what they are. Also we shouldn't mix the "positive" and "negative" forms of checks because that's also confusing to read. `IsSyncSetupComplete` is the "positive" form vs `IsSyncSetupIncomplete`